cmd/kube-apiserver/app/server.go

```diff
@@ -569,7 +575,51 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 	}
 
 	if s.Etcd.EncryptionProviderConfigFilepath != "" {
-		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(s.Etcd.EncryptionProviderConfigFilepath)
+		// Obtain GCE cloud for a Google Cloud KMS connection if needed.
+		// cloudproviders package is in the main kubernetes directory, and thus we cannot access
+		// the gce package (sub-package of cloudproviders) from inside the staging directory.
+		// We need to be able to create a gce cloud object inside the staging directory. Thus, for now,
+		// we pass a function which does that, from here.
+		// If cloudproviders package is moved to a new location, move the logic of this function to kms.Factory.
+		getGCECloud := func() (*kms.GCECloud, error) {
```

Comment:
How are other API servers going to use this?

Our rule of thumb for k8s.io/apiserver is that those servers always run in a Kubernetes environment. That's why we also allow importing k8s.io/client-go from k8s.io/apiserver. I wonder whether the cloud provider is also something belonging either into client-go or into a k8s.io/cloudprovider repo that can be used by any apiserver implementation.

I would prefer to solve this cloudprovider dependency issue before implementing kms with this hack.